modules/google-analytics/classes/wp-google-analytics-legacy.php

```diff
@@ -123,11 +136,43 @@ public function insert_code() {
 					ga.src = ('https:' === document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
 					var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
 				})();
-			</script>\r\n",
+			</script>
+			<!-- End Jetpack Google Analytics -->\r\n",
 			implode( "\r\n", $custom_vars )
 		);
 	}
 
+	/**
+	 * Renders new gtag code.
+	 *
+	 * @param string $tracking_id Google Analytics measurement ID.
+	 */
+	private function render_gtag_code( $tracking_id ) {
+		/**
+		 * Allow for additional elements to be added to the Global Site Tags array.
+		 *
+		 * @since 9.2.0
+		 *
+		 * @param array $universal_commands Array of gtag function calls.
+		 */
+		$universal_commands = apply_filters( 'jetpack_gtag_universal_commands', array() );
+
+		// phpcs:disable WordPress.WP.EnqueuedResources.NonEnqueuedScript
+		?>
+		<!-- Jetpack Google Analytics -->
+		<script async src='https://www.googletagmanager.com/gtag/js?id=<?php echo esc_attr( $tracking_id ); ?>'></script>
+		<script>
+			window.dataLayer = window.dataLayer || [];
+			function gtag() { dataLayer.push( arguments ); }
+			gtag( 'js', new Date() );
+			gtag( 'config', '<?php echo esc_js( $tracking_id ); ?>' );
+			<?php echo esc_js( implode( "\r\n", $universal_commands ) ); ?>
```

Comment:
This also seems wrong. As far as I can tell `$universal_commands` is supposed to be something like
```php
$universal_commands = array(
    "gtag( 'event', 'foo', { 'some': 'data' } );", 
    "gtag( 'event', 'bar', { 'some': 'data' } );", 
);
```
Applying the code you have here would result in invalid JS:
```js
gtag( \'event\', \'foo\', { \'some\': \'data\' } );\ngtag( \'event\', \'bar\', { \'some\': \'data\' } );
```

Personally, I'd probably make `$universal_commands` be more like
```php
$universal_commands = [
    array( 'event', 'foo', array( 'some' => 'data' ) ), 
    array( 'event', 'bar', array( 'some' => 'data' ) ), 
];
```
then have code here turn that into `gtag()` calls with something like
```php
foreach ( $universal_commands as $command ) {
    echo 'gtag( ' . implode( ', ', array_map( 'json_encode', $command ) ) . " );\n";
}
```

Reply:
Done

Reply:
Also the `apply_filters` was wrapped in an `array` so when you tried to add events you ended up with incorrect JS format for `gtag`.

I updated this and it seems to work. See testing instructions I added to the PR desc. 🤞